### PR TITLE
Upgrade Jest and drop Node 4

### PR DESCRIPTION
**Pull request by rogeliog**

### CODE REVIEW

**Comment by SimenB** [comment]
@ljharb yes, which is needed for #43

**Comment by rogeliog** [comment]
@ljharb, it is just a `devDepencency` so it is just the version of Jest used to test this runner.

**Comment by rogeliog** [comment]
I guess it is fine to drop Node 4 now that its end of life has come https://github.com/nodejs/Release cc: @ljharb

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 language: node_js
 node_js:
-  - "8"
-  - "7"
-  - "6"
-  - "4"
+  - '8'
+  - '7'
```
Comment (SimenB) [inline .travis.yml]: 7 is EOL as well.

Should add 10

**Comment by ljharb** [comment]
A node version being EOL means you *could* drop it; in no way does it mean one *should*.

However, jest’s aggressive dropping of node versions might force our hand, as a jest plugin.

**Comment by rogeliog** [comment]
@ljharb I agree 😄 ! As you mentioned, I meant this in the context of Jest already not supporting Node 4 https://github.com/facebook/jest/pull/4769

**Pull request merged**